// mrz/lib/parse/fieldTemplates.js
'use strict';
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.issuingStateTemplate = exports.lastNameTemplate = exports.firstNameTemplate = exports.compositeCheckDigitTemplate = exports.issueDateTemplate = exports.birthDateCheckDigitTemplate = exports.birthDateTemplate = exports.expirationDateCheckDigitTemplate = exports.expirationDateTemplate = exports.sexTemplate = exports.nationalityTemplate = exports.documentCodeTemplate = exports.documentNumberCheckDigitTemplate = exports.documentNumberTemplate = void 0;
const parseCompositeCheckDigit_1 = __importDefault(require("../parsers/parseCompositeCheckDigit"));
const parseDate_1 = __importDefault(require("../parsers/parseDate"));
const parseDateCheckDigit_1 = __importDefault(require("../parsers/parseDateCheckDigit"));
const parseDocumentNumber_1 = __importDefault(require("../parsers/parseDocumentNumber"));
const parseDocumentNumberCheckDigit_1 = __importDefault(require("../parsers/parseDocumentNumberCheckDigit"));
const parseFirstName_1 = __importDefault(require("../parsers/parseFirstName"));
const parseLastName_1 = __importDefault(require("../parsers/parseLastName"));
const parseSex_1 = __importDefault(require("../parsers/parseSex"));
const parseState_1 = __importDefault(require("../parsers/parseState"));
const createFieldParser_1 = require("./createFieldParser");
const documentNumberTemplate = {
    label: 'Document number',
    field: 'documentNumber',
    parser: parseDocumentNumber_1.default,
    type: createFieldParser_1.fieldTypes.ALPHANUMERIC,
};
exports.documentNumberTemplate = documentNumberTemplate;
const documentNumberCheckDigitTemplate = {
    label: 'Document number check digit',
    field: 'documentNumberCheckDigit',
    parser: parseDocumentNumberCheckDigit_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.documentNumberCheckDigitTemplate = documentNumberCheckDigitTemplate;
const documentCodeTemplate = {
    label: 'Document code',
    field: 'documentCode',
    type: createFieldParser_1.fieldTypes.ALPHABETIC,
};
exports.documentCodeTemplate = documentCodeTemplate;
const nationalityTemplate = {
    label: 'Nationality',
    field: 'nationality',
    parser: parseState_1.default,
    type: createFieldParser_1.fieldTypes.ALPHABETIC,
};
exports.nationalityTemplate = nationalityTemplate;
const sexTemplate = {
    label: 'Sex',
    field: 'sex',
    parser: parseSex_1.default,
    type: createFieldParser_1.fieldTypes.ALPHABETIC,
};
exports.sexTemplate = sexTemplate;
const expirationDateTemplate = {
    label: 'Expiration date',
    field: 'expirationDate',
    parser: parseDate_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.expirationDateTemplate = expirationDateTemplate;
const expirationDateCheckDigitTemplate = {
    label: 'Expiration date check digit',
    field: 'expirationDateCheckDigit',
    parser: parseDateCheckDigit_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.expirationDateCheckDigitTemplate = expirationDateCheckDigitTemplate;
const compositeCheckDigitTemplate = {
    label: 'Composite check digit',
    field: 'compositeCheckDigit',
    parser: parseCompositeCheckDigit_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.compositeCheckDigitTemplate = compositeCheckDigitTemplate;
const birthDateTemplate = {
    label: 'Birth date',
    field: 'birthDate',
    parser: parseDate_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.birthDateTemplate = birthDateTemplate;
const birthDateCheckDigitTemplate = {
    label: 'Birth date check digit',
    field: 'birthDateCheckDigit',
    parser: parseDateCheckDigit_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.birthDateCheckDigitTemplate = birthDateCheckDigitTemplate;
const issueDateTemplate = {
    label: 'Issue date',
    field: 'issueDate',
    parser: parseDate_1.default,
    type: createFieldParser_1.fieldTypes.NUMERIC,
};
exports.issueDateTemplate = issueDateTemplate;
const firstNameTemplate = {
    label: 'First name',
    field: 'firstName',
    parser: parseFirstName_1.default,
    type: createFieldParser_1.fieldTypes.ALPHABETIC,
};
exports.firstNameTemplate = firstNameTemplate;
const lastNameTemplate = {
    label: 'Last name',
    field: 'lastName',
    parser: parseLastName_1.default,
    type: createFieldParser_1.fieldTypes.ALPHABETIC,
};
exports.lastNameTemplate = lastNameTemplate;
const issuingStateTemplate = {
    label: 'Issuing state',
    field: 'issuingState',
    parser: parseState_1.default,
    type: createFieldParser_1.fieldTypes.ALPHABETIC,
};
exports.issuingStateTemplate = issuingStateTemplate;
