// mrz/lib/parse/frenchNationalIdFields.js
'use strict';
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const parseAlpha_1 = require("../parsers/parseAlpha");
const parseDocumentCodeId_1 = __importDefault(require("../parsers/parseDocumentCodeId"));
const parseOptional_1 = require("../parsers/parseOptional");
const createFieldParser_1 = __importDefault(require("./createFieldParser"));
const fieldTemplates_1 = require("./fieldTemplates");
exports.default = [
    {
        ...fieldTemplates_1.documentCodeTemplate,
        line: 0,
        start: 0,
        end: 2,
        parser: parseDocumentCodeId_1.default,
    },
    { ...fieldTemplates_1.issuingStateTemplate, line: 0, start: 2, end: 5 },
    { ...fieldTemplates_1.lastNameTemplate, line: 0, start: 5, end: 30, parser: parseAlpha_1.parseAlpha },
    {
        label: 'Administrative code',
        field: 'administrativeCode',
        line: 0,
        start: 30,
        end: 36,
        parser: parseOptional_1.parseOptional,
    },
    { ...fieldTemplates_1.issueDateTemplate, line: 1, start: 0, end: 4 },
    {
        label: 'Administrative code 2',
        field: 'administrativeCode2',
        line: 1,
        start: 4,
        end: 7,
        parser: parseOptional_1.parseOptional,
    },
    { ...fieldTemplates_1.documentNumberTemplate, line: 1, start: 7, end: 12 },
    {
        ...fieldTemplates_1.documentNumberCheckDigitTemplate,
        line: 1,
        start: 12,
        end: 13,
        related: [
            {
                line: 1,
                start: 0,
                end: 12,
            },
        ],
    },
    { ...fieldTemplates_1.firstNameTemplate, line: 1, start: 13, end: 27, parser: parseAlpha_1.parseAlpha },
    { ...fieldTemplates_1.birthDateTemplate, line: 1, start: 27, end: 33 },
    {
        ...fieldTemplates_1.birthDateCheckDigitTemplate,
        line: 1,
        start: 33,
        end: 34,
        related: [
            {
                line: 1,
                start: 27,
                end: 33,
            },
        ],
    },
    { ...fieldTemplates_1.sexTemplate, line: 1, start: 34, end: 35 },
    {
        ...fieldTemplates_1.compositeCheckDigitTemplate,
        line: 1,
        start: 35,
        end: 36,
        related: [
            {
                line: 0,
                start: 0,
                end: 36,
            },
            {
                line: 1,
                start: 0,
                end: 35,
            },
        ],
    },
].map(createFieldParser_1.default);
