// mrz/lib/parse/getResult.js
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.getResult = void 0;
function getDetails(lines, fieldParsers, autocorrectArray) {
    const details = [];
    fieldParsers.forEach(({ parser }, i) => {
        details.push(parser(lines, autocorrectArray[i]));
    });
    return details;
}
function getFields(details) {
    const fields = {};
    let valid = true;
    for (const detail of details) {
        if (!detail.valid)
            valid = false;
        if (detail.field) {
            fields[detail.field] = detail.value;
        }
    }
    return { fields, valid };
}
function getCorrection(lines, fieldParsers, autocorrect) {
    const corrected = lines.slice();
    let autocorrectArray = [];
    if (autocorrect) {
        fieldParsers.forEach(({ autocorrector }) => {
            const { autocorrect, correctedText, range } = autocorrector(lines);
            autocorrectArray.push(autocorrect);
            const line = corrected[range.line];
            corrected[range.line] =
                line.slice(0, range.start) + correctedText + line.slice(range.end);
        });
    }
    else {
        autocorrectArray = new Array(fieldParsers.length).fill([]);
    }
    return { corrected, autocorrectArray };
}
function getResult(format, lines, fieldParsers, options) {
    const { autocorrect = false } = options;
    const { corrected, autocorrectArray } = getCorrection(lines, fieldParsers, autocorrect);
    const details = getDetails(corrected, fieldParsers, autocorrectArray);
    const fields = getFields(details);
    return {
        format,
        details,
        fields: fields.fields,
        valid: fields.valid,
    };
}
exports.getResult = getResult;
