// mrz/lib/parse/swissDrivingLicenseFields.js
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const parseNumber_1 = require("../parsers/parseNumber");
const checkSeparator_1 = require("../parsers/swissDrivingLicense/checkSeparator");
const parseDocumentCode_1 = __importDefault(require("../parsers/swissDrivingLicense/parseDocumentCode"));
const parseDocumentNumber_1 = __importDefault(require("../parsers/swissDrivingLicense/parseDocumentNumber"));
const parseIssuingState_1 = __importDefault(require("../parsers/swissDrivingLicense/parseIssuingState"));
const parseLanguageCode_1 = __importDefault(require("../parsers/swissDrivingLicense/parseLanguageCode"));
const createFieldParser_1 = __importStar(require("./createFieldParser"));
const fieldTemplates_1 = require("./fieldTemplates");
exports.default = [
    {
        ...fieldTemplates_1.documentNumberTemplate,
        line: 0,
        start: 0,
        end: 9,
        parser: parseDocumentNumber_1.default,
    },
    {
        label: 'Language code',
        field: 'languageCode',
        line: 0,
        start: 6,
        end: 7,
        parser: parseLanguageCode_1.default,
        type: createFieldParser_1.fieldTypes.ALPHABETIC,
    },
    {
        ...fieldTemplates_1.documentCodeTemplate,
        line: 1,
        start: 0,
        end: 2,
        parser: parseDocumentCode_1.default,
    },
    {
        ...fieldTemplates_1.issuingStateTemplate,
        line: 1,
        start: 2,
        end: 5,
        parser: parseIssuingState_1.default,
    },
    {
        label: 'PIN code',
        field: 'pinCode',
        line: 1,
        start: 5,
        end: 14,
        parser: parseNumber_1.parseNumber,
        type: createFieldParser_1.fieldTypes.NUMERIC,
    },
    {
        label: 'Version number',
        field: 'versionNumber',
        line: 1,
        start: 14,
        end: 17,
        parser: parseNumber_1.parseNumber,
        type: createFieldParser_1.fieldTypes.NUMERIC,
    },
    {
        label: 'Separator 1',
        field: null,
        line: 1,
        start: 17,
        end: 19,
        parser: checkSeparator_1.checkSeparator,
    },
    { ...fieldTemplates_1.birthDateTemplate, line: 1, start: 19, end: 25 },
    {
        label: 'Separator 2',
        field: null,
        line: 1,
        start: 25,
        end: 30,
        parser: checkSeparator_1.checkSeparator,
    },
    { ...fieldTemplates_1.lastNameTemplate, line: 2, start: 0, end: 30 },
    { ...fieldTemplates_1.firstNameTemplate, line: 2, start: 0, end: 30 },
].map(createFieldParser_1.default);
