// mrz/lib/parse/td1Fields.js
'use strict';
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const parseDocumentCodeId_1 = __importDefault(require("../parsers/parseDocumentCodeId"));
const parseDocumentNumberOptional_1 = require("../parsers/parseDocumentNumberOptional");
const parseOptional_1 = require("../parsers/parseOptional");
const createFieldParser_1 = __importDefault(require("./createFieldParser"));
const fieldTemplates_1 = require("./fieldTemplates");
exports.default = [
    {
        ...fieldTemplates_1.documentCodeTemplate,
        line: 0,
        start: 0,
        end: 2,
        parser: parseDocumentCodeId_1.default,
    },
    { ...fieldTemplates_1.issuingStateTemplate, line: 0, start: 2, end: 5 },
    {
        ...fieldTemplates_1.documentNumberTemplate,
        line: 0,
        start: 5,
        end: 14,
        related: [
            {
                line: 0,
                start: 14,
                end: 15,
            },
            {
                line: 0,
                start: 15,
                end: 30,
            },
        ],
    },
    {
        ...fieldTemplates_1.documentNumberCheckDigitTemplate,
        line: 0,
        start: 14,
        end: 15,
        related: [
            {
                line: 0,
                start: 5,
                end: 14,
            },
            {
                line: 0,
                start: 15,
                end: 30,
            },
        ],
    },
    {
        label: 'Optional field 1',
        field: 'optional1',
        line: 0,
        start: 15,
        end: 30,
        related: [
            {
                line: 0,
                start: 5,
                end: 14,
            },
            {
                line: 0,
                start: 14,
                end: 15,
            },
        ],
        parser: parseDocumentNumberOptional_1.parseDocumentNumberOptional,
    },
    { ...fieldTemplates_1.birthDateTemplate, start: 0, end: 6, line: 1 },
    {
        ...fieldTemplates_1.birthDateCheckDigitTemplate,
        line: 1,
        start: 6,
        end: 7,
        related: [
            {
                line: 1,
                start: 0,
                end: 6,
            },
        ],
    },
    { ...fieldTemplates_1.sexTemplate, line: 1, start: 7, end: 8 },
    { ...fieldTemplates_1.expirationDateTemplate, line: 1, start: 8, end: 14 },
    {
        ...fieldTemplates_1.expirationDateCheckDigitTemplate,
        line: 1,
        start: 14,
        end: 15,
        related: [
            {
                line: 1,
                start: 8,
                end: 14,
            },
        ],
    },
    { ...fieldTemplates_1.nationalityTemplate, line: 1, start: 15, end: 18 },
    {
        label: 'Optional field 2',
        field: 'optional2',
        line: 1,
        start: 18,
        end: 29,
        parser: parseOptional_1.parseOptional,
    },
    {
        ...fieldTemplates_1.compositeCheckDigitTemplate,
        line: 1,
        start: 29,
        end: 30,
        related: [
            {
                line: 0,
                start: 5,
                end: 30,
            },
            {
                line: 1,
                start: 0,
                end: 7,
            },
            {
                line: 1,
                start: 8,
                end: 15,
            },
            {
                line: 1,
                start: 18,
                end: 29,
            },
        ],
    },
    { ...fieldTemplates_1.lastNameTemplate, line: 2, start: 0, end: 30 },
    { ...fieldTemplates_1.firstNameTemplate, line: 2, start: 0, end: 30 },
].map(createFieldParser_1.default);
