// mrz/lib/parse/td3Fields.js
'use strict';
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const parseDocumentCodePassport_1 = __importDefault(require("../parsers/parseDocumentCodePassport"));
const parsePersonalNumber_1 = require("../parsers/parsePersonalNumber");
const parsePersonalNumberCheckDigit_1 = require("../parsers/parsePersonalNumberCheckDigit");
const createFieldParser_1 = __importDefault(require("./createFieldParser"));
const fieldTemplates_1 = require("./fieldTemplates");
exports.default = [
    {
        ...fieldTemplates_1.documentCodeTemplate,
        line: 0,
        start: 0,
        end: 2,
        parser: parseDocumentCodePassport_1.default,
    },
    { ...fieldTemplates_1.issuingStateTemplate, line: 0, start: 2, end: 5 },
    { ...fieldTemplates_1.lastNameTemplate, line: 0, start: 5, end: 44 },
    { ...fieldTemplates_1.firstNameTemplate, line: 0, start: 5, end: 44 },
    { ...fieldTemplates_1.documentNumberTemplate, line: 1, start: 0, end: 9 },
    {
        ...fieldTemplates_1.documentNumberCheckDigitTemplate,
        line: 1,
        start: 9,
        end: 10,
        related: [
            {
                line: 1,
                start: 0,
                end: 9,
            },
        ],
    },
    { ...fieldTemplates_1.nationalityTemplate, line: 1, start: 10, end: 13 },
    { ...fieldTemplates_1.birthDateTemplate, line: 1, start: 13, end: 19 },
    {
        ...fieldTemplates_1.birthDateCheckDigitTemplate,
        line: 1,
        start: 19,
        end: 20,
        related: [
            {
                line: 1,
                start: 13,
                end: 19,
            },
        ],
    },
    { ...fieldTemplates_1.sexTemplate, line: 1, start: 20, end: 21 },
    { ...fieldTemplates_1.expirationDateTemplate, line: 1, start: 21, end: 27 },
    {
        ...fieldTemplates_1.expirationDateCheckDigitTemplate,
        line: 1,
        start: 27,
        end: 28,
        related: [
            {
                line: 1,
                start: 21,
                end: 27,
            },
        ],
    },
    {
        label: 'Personal number',
        field: 'personalNumber',
        line: 1,
        start: 28,
        end: 42,
        parser: parsePersonalNumber_1.parsePersonalNumber,
    },
    {
        label: 'Personal number check digit',
        field: 'personalNumberCheckDigit',
        line: 1,
        start: 42,
        end: 43,
        related: [
            {
                line: 1,
                start: 28,
                end: 42,
            },
        ],
        parser: parsePersonalNumberCheckDigit_1.parsePersonalNumberCheckDigit,
    },
    {
        ...fieldTemplates_1.compositeCheckDigitTemplate,
        line: 1,
        start: 43,
        end: 44,
        related: [
            {
                line: 1,
                start: 0,
                end: 10,
            },
            {
                line: 1,
                start: 13,
                end: 20,
            },
            {
                line: 1,
                start: 21,
                end: 43,
            },
        ],
    },
].map(createFieldParser_1.default);
